data-prepper-core/src/test/java/org/opensearch/dataprepper/core/peerforwarder/PeerForwardingProcessingDecoratorTest.java

```diff
@@ -171,6 +171,45 @@ void PeerForwardingProcessingDecorator_should_have_interaction_with_getIdentific
             verifyNoMoreInteractions(peerForwarderProvider);
         }
 
+        @Test
+        void PeerForwardingProcessingDecorator_with_localProcessingOnlyWithExcludeIdentificationKeys() {
+            List<Processor> processorList = new ArrayList<>();
+            processorList.add((Processor) requiresPeerForwarding);
+            processorList.add((Processor) requiresPeerForwardingCopy);
+
+            LocalPeerForwarder localPeerForwarder = mock(LocalPeerForwarder.class);
+            when(peerForwarderProvider.register(pipelineName, (Processor) requiresPeerForwarding, pluginId, identificationKeys, PIPELINE_WORKER_THREADS)).thenReturn(localPeerForwarder);
+            Event event = mock(Event.class);
+            when(record.getData()).thenReturn(event);
+            List<Record<Event>> testData = Collections.singletonList(record);
+            when(requiresPeerForwarding.isApplicableEventForPeerForwarding(event)).thenReturn(true);
+            when(requiresPeerForwardingCopy.isApplicableEventForPeerForwarding(event)).thenReturn(true);
+
+            Processor processor1 = (Processor)requiresPeerForwarding;
+            Processor processor2 = (Processor)requiresPeerForwardingCopy;
+            when(processor1.execute(testData)).thenReturn(testData);
+            when(processor2.execute(testData)).thenReturn(testData);
+
+            when(requiresPeerForwarding.getIdentificationKeys()).thenReturn(identificationKeys);
+            when(requiresPeerForwarding.getIdentificationKeys()).thenReturn(identificationKeys);
+            when(requiresPeerForwardingCopy.getIdentificationKeys()).thenReturn(identificationKeys);
+
+            final List<Processor> processors = createObjectUnderTestDecoratedProcessorsWithExcludeIdentificationKeys(processorList, Set.of(identificationKeys));
+            assertThat(processors.size(), equalTo(2));
+            for (final Processor processor: processors) {
+                assertTrue(((PeerForwardingProcessorDecorator)processor).isPeerForwardingDisabled());
+            }
+            verify(peerForwarderProvider, times(1)).register(pipelineName, processor, pluginId, identificationKeys, PIPELINE_WORKER_THREADS);
+            verifyNoMoreInteractions(peerForwarderProvider);
+            Collection<Record<Event>> result = processors.get(0).execute(testData);
```

Comment:
I'm not following this test. It seems that you mock this above on line 190 and then assert it here.

I'm not 100% confident that this test is covering the situation.

Reply:
OK. I think I see now. The `processors` are the decorated processors under test.

Maybe rename to `objectsUnderTest`.